File: jpa-example/src/main/java/example/jpa/model/Tag.java
package example.jpa.model;

import jakarta.persistence.Entity;
import jakarta.persistence.ManyToMany;
import lombok.Getter;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@Entity
public class Tag extends SerialId {
    public Tag() {
    }

    public Tag(String name) {
        this.name = name;
    }

    private String name;

    @ManyToMany(mappedBy = "tags")
    private Set<Book> books = new HashSet<>();
}

File: jimmer-example/src/main/kotlin/example/jimmer/model/Tag.kt
package example.jimmer.model

import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.Key
import org.babyfish.jimmer.sql.KeyUniqueConstraint
import org.babyfish.jimmer.sql.ManyToMany

@Entity
@KeyUniqueConstraint
interface Tag : SerialId {
    @Key
    val name: String

    @ManyToMany(mappedBy = "tags")
    val books: List<Book>

}

fun Tag(name: String): Tag = Tag { this.name = name }
